test/integration/scheduler_perf/util.go

```diff
@@ -180,28 +180,25 @@ func (pc *metricsCollector) collect() []DataItem {
 }
 
 func collectHistogram(metric string, labels map[string]string) *DataItem {
-	hist, err := testutil.GetHistogramFromGatherer(legacyregistry.DefaultGatherer, metric)
+	vec, err := testutil.GetHistogramVecFromGatherer(legacyregistry.DefaultGatherer, metric, nil)
 	if err != nil {
 		klog.Error(err)
 		return nil
 	}
-	if hist.Histogram == nil {
```

Comment:
With vector of histograms, one can get an empty vector. Which will produce NaN for quantiles. Though, when there's one histogram with no buckets, one gets NaN anyway.

Reply:
How about avoid adding nil Histogram? (although I guess it won't be generated in Promotheus side, but just for double check)

Reply:
Sounds good. I see you already added the check for it as well. Awesome!!!